In [9]:
import pandas as pd

In [19]:
data = pd.read_csv('data.csv')
data

id kind     code           class_name prof_name  room date1 date2  \
0      0   교필  AK00055  비아메디아채플(나와 타인 이해하기)       이남호  1505     월   NaN   
1      1   교필  AK00055     비아메디아채플(리더십의 영성)       김한승  3501     월   NaN   
2      2   교필  AK00055       비아메디아채플(서양음악사)       박태영  M301     월   NaN   
3      3   교필  AK00055      비아메디아채플(성공회이야기)       노철래  3301     목   NaN   
4      4   교필  AK00055      비아메디아채플(시각과 영성)       김돈회  M406     월   NaN   
..   ...  ...      ...                  ...       ...   ...   ...   ...   
413  413   전선  XE00003             데이터기초수학2       백지현  6206     화     목   
414  414   전선  XE00004               고급머신러닝       홍성준  6405     목   NaN   
415  415   전선  XE00005        고급Python프로그래밍       홍성준  6405     화   NaN   
416  416   전선  XE00006             딥러닝프레임워크       이상윤  6405     월     수   
417  417   전선  XE00007              확률과랜덤변수       이상윤  6405     월     수   

     start    end  
0    11:00  11:50  
1    11:00  11:50  
2    11:00  11:50  
3    11:00  11:50  
4    11:00  11:50  
..     ...    ...  
413  12:00  13:15  
414  09:00  11:50  
415  09:00  11:50  
416  13:30  14:45  
417  16:30  17:45  

[418 rows x 10 columns]

In [20]:
data.info()
# 아니 room에 null이 있네?

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 418 entries, 0 to 417
Data columns (total 10 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   id          418 non-null    int64 
 1   kind        418 non-null    object
 2   code        418 non-null    object
 3   class_name  418 non-null    object
 4   prof_name   418 non-null    object
 5   room        416 non-null    object
 6   date1       414 non-null    object
 7   date2       105 non-null    object
 8   start       411 non-null    object
 9   end         411 non-null    object
dtypes: int64(1), object(9)
memory usage: 32.8+ KB


In [22]:
data[data['room'].isnull()]


id kind     code class_name prof_name room date1 date2 start  end
143  143   교선  AE00029     사회봉사II       노철래  NaN   NaN   NaN   NaN  NaN
144  144   교선  AE00030    사회봉사III       노철래  NaN   NaN   NaN   NaN  NaN

In [23]:
data[data['date1'].isnull()]


id kind     code class_name prof_name  room date1 date2 start  end
12    12   교필  AC00003      사회봉사I       노철래  7207   NaN   NaN   NaN  NaN
13    13   교필  AC00003      사회봉사I       노철래  7207   NaN   NaN   NaN  NaN
143  143   교선  AE00029     사회봉사II       노철래   NaN   NaN   NaN   NaN  NaN
144  144   교선  AE00030    사회봉사III       노철래   NaN   NaN   NaN   NaN  NaN

pdf 확인해보니까 사회봉사는 교필과 교선이 존재함.

교필은 7207로 강의실이 있는데 강의실/시간이 '/'로 구분되어 있지 않음. 
그래서 date1, start, end가 적용되지 않았음 -> 직접 입력해주자.

교선 사회봉사는 pdf에서부터 강의실/시간이 비어있음. 이건 그냥 두면 될듯.

In [26]:
## 교필 사봉의 date, start, end 값 변경

data.at[143, 'date1'] = '수'
data.at[143, 'start'] = '10:30'
data.at[143, 'end'] = '11:45'

data.at[144, 'date1'] = '목'
data.at[144, 'start'] = '10:30'
data.at[144, 'end'] = '11:45'

data[143:145]

id kind     code class_name prof_name room date1 date2  start    end
143  143   교선  AE00029     사회봉사II       노철래  NaN     수   NaN  10:30  11:45
144  144   교선  AE00030    사회봉사III       노철래  NaN     목   NaN  10:30  11:45

In [28]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 418 entries, 0 to 417
Data columns (total 10 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   id          418 non-null    int64 
 1   kind        418 non-null    object
 2   code        418 non-null    object
 3   class_name  418 non-null    object
 4   prof_name   418 non-null    object
 5   room        416 non-null    object
 6   date1       416 non-null    object
 7   date2       105 non-null    object
 8   start       413 non-null    object
 9   end         413 non-null    object
dtypes: int64(1), object(9)
memory usage: 32.8+ KB


start와 end가 없는 것도 있네

In [30]:
data[data['start'].isnull()]

id kind     code  class_name prof_name  room         date1 date2 start  \
12    12   교필  AC00003       사회봉사I       노철래  7207           NaN   NaN   NaN   
13    13   교필  AC00003       사회봉사I       노철래  7207           NaN   NaN   NaN   
168  168   전탐  LA00028  과학카페에서읽는종교       신익상  7208  목09:00-11:50   NaN   NaN   
229  229   전선  AD00093    사이버세계와종교       김힌나  3402  수15:00~17:50   NaN   NaN   
230  230   전선  AE00104   대안경제와생태사회       신익상  3301  목12:00~14:50   NaN   NaN   

     end  
12   NaN  
13   NaN  
168  NaN  
229  NaN  
230  NaN

168번만 왜 시간을 저런식으로 써둔거지 어쨋든 나머지도 다 수정해주자

In [35]:
data.at[168, 'date1'] = '목'
data.at[168, 'start'] = '09:00'
data.at[168, 'end'] = '11:50'

data.at[229, 'date1'] = '수'
data.at[229, 'start'] = '15:00'
data.at[229, 'end'] = '17:50'

data.at[230, 'date1'] = '목'
data.at[230, 'start'] = '12:00'
data.at[230, 'end'] = '14:50'

data[229:231]

id kind     code class_name prof_name  room date1 date2  start    end
229  229   전선  AD00093   사이버세계와종교       김힌나  3402     수   NaN  15:00  17:50
230  230   전선  AE00104  대안경제와생태사회       신익상  3301     목   NaN  12:00  14:50

In [37]:
data.info()
# 이제 진짜 전처리 완료..

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 418 entries, 0 to 417
Data columns (total 10 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   id          418 non-null    int64 
 1   kind        418 non-null    object
 2   code        418 non-null    object
 3   class_name  418 non-null    object
 4   prof_name   418 non-null    object
 5   room        416 non-null    object
 6   date1       416 non-null    object
 7   date2       105 non-null    object
 8   start       416 non-null    object
 9   end         416 non-null    object
dtypes: int64(1), object(9)
memory usage: 32.8+ KB


In [79]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 418 entries, 0 to 417
Data columns (total 11 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   id          417 non-null    float64
 1   kind        418 non-null    object 
 2   code        418 non-null    object 
 3   class_name  418 non-null    object 
 4   prof_name   418 non-null    object 
 5   room        416 non-null    object 
 6   date1       416 non-null    object 
 7   date2       105 non-null    object 
 8   start       416 non-null    object 
 9   end         416 non-null    object 
 10  kwan        418 non-null    object 
dtypes: float64(1), object(10)
memory usage: 36.0+ KB


In [38]:
data.columns

Index(['id', 'kind', 'code', 'class_name', 'prof_name', 'room', 'date1',
       'date2', 'start', 'end'],
      dtype='object')

In [45]:
## 각 행에 관 컬럼을 추가할 것이다.
## 어떤강의실 있는지 보기 위해 강의실 중복 제거

df = data
df['kwan'] = data.room.str[:1]
df.drop_duplicates(['kwan'])

id kind     code             class_name prof_name   room date1 date2  \
0      0   교필  AK00055    비아메디아채플(나와 타인 이해하기)       이남호   1505     월   NaN   
1      1   교필  AK00055       비아메디아채플(리더십의 영성)       김한승   3501     월   NaN   
2      2   교필  AK00055         비아메디아채플(서양음악사)       박태영   M301     월   NaN   
5      5   교필  AK00055     비아메디아채플(열린 성공회 예배)       미지정   9301     목   NaN   
8      8   교필  AK00055  비아메디아채플(인간심리와 기독교 명상)       이종민   7205     목   NaN   
34    34   교필  AC00012      대학생활세미나 I(IT융합1반)       김선형   6201     금   NaN   
66    66   교필  AC00015           데이터 활용 및  분석       이해신   B105     금   NaN   
135  135   교선  AD00164                 생활과 발레       장소정  5C301     목   NaN   
143  143   교선  AE00029                 사회봉사II       노철래    NaN     수   NaN   

     start    end kwan  
0    11:00  11:50    1  
1    11:00  11:50    3  
2    11:00  11:50    M  
5    11:00  11:50    9  
8    11:00  11:50    7  
34   10:00  11:50    6  
66   10:00  11:50    B  
135  10:00  11:50    5  
143  10:30  11:45  NaN

In [46]:
conditionlist = [
    (df['kwan'] == '1'),
    (df['kwan'] == '3'), 
    (df['kwan'] == 'M'),
    (df['kwan'] == '9'),
    (df['kwan'] == '7'),
    (df['kwan'] == '6'),
    (df['kwan'] == 'B'),
    (df['kwan'] == '5')
]
choicelist = [
    '승연관',
    '월당관',
    '미가엘관',
    '성미가엘성당&피츠버그홀',
    '새천년관',
    '정보과학관',
    '일만관',
    '나눔관'
]
import numpy as np
df['kwan'] = np.select(conditionlist, choicelist, default=np.nan)

C:\Users\mirae\AppData\Local\Temp\ipykernel_18640\3945067208.py:22: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df['kwan'] = np.select(conditionlist, choicelist, default=np.nan)


In [47]:
df

id kind     code             class_name prof_name   room date1 date2  \
0      0   교필  AK00055    비아메디아채플(나와 타인 이해하기)       이남호   1505     월   NaN   
1      1   교필  AK00055       비아메디아채플(리더십의 영성)       김한승   3501     월   NaN   
2      2   교필  AK00055         비아메디아채플(서양음악사)       박태영   M301     월   NaN   
5      5   교필  AK00055     비아메디아채플(열린 성공회 예배)       미지정   9301     목   NaN   
8      8   교필  AK00055  비아메디아채플(인간심리와 기독교 명상)       이종민   7205     목   NaN   
34    34   교필  AC00012      대학생활세미나 I(IT융합1반)       김선형   6201     금   NaN   
66    66   교필  AC00015           데이터 활용 및  분석       이해신   B105     금   NaN   
135  135   교선  AD00164                 생활과 발레       장소정  5C301     목   NaN   
143  143   교선  AE00029                 사회봉사II       노철래    NaN     수   NaN   

     start    end          kwan  
0    11:00  11:50           승연관  
1    11:00  11:50           월당관  
2    11:00  11:50          미가엘관  
5    11:00  11:50  성미가엘성당&피츠버그홀  
8    11:00  11:50          새천년관  
34   10:00  11:50         정보과학관  
66   10:00  11:50           일만관  
135  10:00  11:50           나눔관  
143  10:30  11:45           nan

In [40]:
data

id kind     code           class_name prof_name  room date1 date2  \
0      0   교필  AK00055  비아메디아채플(나와 타인 이해하기)       이남호  1505     월   NaN   
1      1   교필  AK00055     비아메디아채플(리더십의 영성)       김한승  3501     월   NaN   
2      2   교필  AK00055       비아메디아채플(서양음악사)       박태영  M301     월   NaN   
3      3   교필  AK00055      비아메디아채플(성공회이야기)       노철래  3301     목   NaN   
4      4   교필  AK00055      비아메디아채플(시각과 영성)       김돈회  M406     월   NaN   
..   ...  ...      ...                  ...       ...   ...   ...   ...   
413  413   전선  XE00003             데이터기초수학2       백지현  6206     화     목   
414  414   전선  XE00004               고급머신러닝       홍성준  6405     목   NaN   
415  415   전선  XE00005        고급Python프로그래밍       홍성준  6405     화   NaN   
416  416   전선  XE00006             딥러닝프레임워크       이상윤  6405     월     수   
417  417   전선  XE00007              확률과랜덤변수       이상윤  6405     월     수   

     start    end  
0    11:00  11:50  
1    11:00  11:50  
2    11:00  11:50  
3    11:00  11:50  
4    11:00  11:50  
..     ...    ...  
413  12:00  13:15  
414  09:00  11:50  
415  09:00  11:50  
416  13:30  14:45  
417  16:30  17:45  

[418 rows x 10 columns]

In [48]:
## 적용해보기

data['kwan'] = data.room.str[:1]

In [49]:
import numpy as np

In [50]:
## data에서 room의 앞이 7이면 새 변수(kwan)에 새천년관 넣기 - 조건문

conditionlist = [
    (data['kwan'] == '1'),
    (data['kwan'] == '3'), 
    (data['kwan'] == 'M'),
    (data['kwan'] == '9'),
    (data['kwan'] == '7'),
    (data['kwan'] == '6'),
    (data['kwan'] == 'B'),
    (data['kwan'] == '5')
]
choicelist = [
    '승연관',
    '월당관',
    '미가엘관',
    '성미가엘성당&피츠버그홀',
    '새천년관',
    '정보과학관',
    '일만관',
    '나눔관'
]

data['kwan'] = np.select(conditionlist, choicelist, default=np.nan)
data

id kind     code           class_name prof_name  room date1 date2  \
0      0   교필  AK00055  비아메디아채플(나와 타인 이해하기)       이남호  1505     월   NaN   
1      1   교필  AK00055     비아메디아채플(리더십의 영성)       김한승  3501     월   NaN   
2      2   교필  AK00055       비아메디아채플(서양음악사)       박태영  M301     월   NaN   
3      3   교필  AK00055      비아메디아채플(성공회이야기)       노철래  3301     목   NaN   
4      4   교필  AK00055      비아메디아채플(시각과 영성)       김돈회  M406     월   NaN   
..   ...  ...      ...                  ...       ...   ...   ...   ...   
413  413   전선  XE00003             데이터기초수학2       백지현  6206     화     목   
414  414   전선  XE00004               고급머신러닝       홍성준  6405     목   NaN   
415  415   전선  XE00005        고급Python프로그래밍       홍성준  6405     화   NaN   
416  416   전선  XE00006             딥러닝프레임워크       이상윤  6405     월     수   
417  417   전선  XE00007              확률과랜덤변수       이상윤  6405     월     수   

     start    end   kwan  
0    11:00  11:50    승연관  
1    11:00  11:50    월당관  
2    11:00  11:50   미가엘관  
3    11:00  11:50    월당관  
4    11:00  11:50   미가엘관  
..     ...    ...    ...  
413  12:00  13:15  정보과학관  
414  09:00  11:50  정보과학관  
415  09:00  11:50  정보과학관  
416  13:30  14:45  정보과학관  
417  16:30  17:45  정보과학관  

[418 rows x 11 columns]

In [51]:
data.info()
#잘 적용됨

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 418 entries, 0 to 417
Data columns (total 11 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   id          418 non-null    int64 
 1   kind        418 non-null    object
 2   code        418 non-null    object
 3   class_name  418 non-null    object
 4   prof_name   418 non-null    object
 5   room        416 non-null    object
 6   date1       416 non-null    object
 7   date2       105 non-null    object
 8   start       416 non-null    object
 9   end         416 non-null    object
 10  kwan        418 non-null    object
dtypes: int64(1), object(10)
memory usage: 36.0+ KB


In [59]:
## kwan class 생성

kwan = data.drop_duplicates(['kwan'])
kwan = kwan[['kwan']]
kwan = kwan.reset_index()
kwan = kwan[['kwan']]
kwan

kwan
0           승연관
1           월당관
2          미가엘관
3  성미가엘성당&피츠버그홀
4          새천년관
5         정보과학관
6           일만관
7           나눔관
8           nan

In [62]:
kwan.drop(8, axis=0, inplace=True)

KeyError: '[8] not found in axis'

In [63]:
kwan

kwan
0           승연관
1           월당관
2          미가엘관
3  성미가엘성당&피츠버그홀
4          새천년관
5         정보과학관
6           일만관
7           나눔관

In [64]:
kwan.loc[8] = ['중앙도서관']

In [65]:
kwan

kwan
0           승연관
1           월당관
2          미가엘관
3  성미가엘성당&피츠버그홀
4          새천년관
5         정보과학관
6           일만관
7           나눔관
8         중앙도서관

In [66]:
## 관 번호 삽입
kwan['kwan_num'] = ['1','3','M','9','7','6','B','5','8']
kwan

kwan kwan_num
0           승연관        1
1           월당관        3
2          미가엘관        M
3  성미가엘성당&피츠버그홀        9
4          새천년관        7
5         정보과학관        6
6           일만관        B
7           나눔관        5
8         중앙도서관        8

In [67]:
## 열 순서 변경

kwan = kwan[['kwan_num', 'kwan']]
kwan

kwan_num          kwan
0        1           승연관
1        3           월당관
2        M          미가엘관
3        9  성미가엘성당&피츠버그홀
4        7          새천년관
5        6         정보과학관
6        B           일만관
7        5           나눔관
8        8         중앙도서관

In [68]:
## kwan class 생성

kwan.to_csv("kwan.csv", encoding='utf-8-sig')

In [69]:
room

id kind     code                    class_name prof_name   room date1  \
0      0   교필  AK00055           비아메디아채플(나와 타인 이해하기)       이남호   1505     월   
1      1   교필  AK00055              비아메디아채플(리더십의 영성)       김한승   3501     월   
2      2   교필  AK00055                비아메디아채플(서양음악사)       박태영   M301     월   
3      3   교필  AK00055               비아메디아채플(성공회이야기)       노철래   3301     목   
4      4   교필  AK00055               비아메디아채플(시각과 영성)       김돈회   M406     월   
5      5   교필  AK00055            비아메디아채플(열린 성공회 예배)       미지정   9301     목   
6      6   교필  AK00055             비아메디아채플(예술치료와 영성)       이현지   M407     월   
8      8   교필  AK00055         비아메디아채플(인간심리와 기독교 명상)       이종민   7205     목   
9      9   교필  AK00055       비아메디아채플(쟁점으로 만나는 그리스도교)       김종훈   3401     목   
10    10   교필  AK00055  비아메디아채플(정신분석으로 바라본 애착관계와 영성)       김미희   1503     목   
11    11   교필  AK00055            비아메디아채플(함께 나누는 세상)       양만호   M402     월   
12    12   교필  AC00003                         사회봉사I       노철래   7207   NaN   
15    15   교필  AC00012            대학생활세미나 I(미디어융합2반)       박남기   3402     금   
18    18   교필  AC00012             대학생활세미나 I(사회융합1반)       송재민   7204     금   
20    20   교필  AC00012             대학생활세미나 I(사회융합3반)       남일성   7206     금   
21    21   교필  AC00012             대학생활세미나 I(사회융합4반)       차피득   7202     금   
26    26   교필  AC00012             대학생활세미나 I(인문융합1반)       김혜인   7304     금   
27    27   교필  AC00012             대학생활세미나 I(인문융합2반)       이남주   7305     금   
28    28   교필  AC00012             대학생활세미나 I(인문융합3반)       이주엽   7306     금   
29    29   교필  AC00012             대학생활세미나 I(인문융합4반)       미지정   7307     금   
34    34   교필  AC00012             대학생활세미나 I(IT융합1반)       김선형   6201     금   
35    35   교필  AC00012             대학생활세미나 I(IT융합2반)       이하규   6404     금   
36    36   교필  AC00012             대학생활세미나 I(IT융합3반)       정연식   6109     금   
37    37   교필  AC00012             대학생활세미나 I(IT융합4반)       이상윤   6405     금   
42    42   교필  AC00013                           말과글       오현화   1504     수   
52    52   교필  AC00007                         인권과평화       이남주   7208     화   
53    53   교필  AC00007                         인권과평화       양경은   7308     화   
54    54   교필  AC00007                         인권과평화       조경희   7309     화   
62    62   교필  AC00015                  데이터 활용 및  분석       윤명호   M401     금   
64    64   교필  AC00015                  데이터 활용 및  분석       유상신   3302     금   
66    66   교필  AC00015                  데이터 활용 및  분석       이해신   B105     금   
70    70   교선  AK00025                       현대사회와법1       이기덕   3502     화   
74    74   교선  AE00019                      국제질서의 이해       이남주   6110     화   
77    77   교선  AF00056                기후위기와 지속가능한 농업       김병수   M403     목   
85    85   교선  AD00172         더불어숲인문학\n(삶을 위한 인문공부)       고병헌   7301     월   
95    95   교선  AH00020                    진로와 직업의 세계       한상진   7303     화   
109  109   교선  AE00099                            마음       남일성   9101     화   
135  135   교선  AD00164                        생활과 발레       장소정  5C301     목   
143  143   교선  AE00029                        사회봉사II       노철래    NaN     수   
145  145   평교  AD00124                       문화예술교육론       신재민   1502     금   
179  179   전탐  VS00012                    경제환경과 기업경영        이웅   M205     화   
191  191   전탐  VS00020                  전공탐색세미나(경제학)       유철규   7302     월   
197  197   전탐  VD00003                      디지털기초디자인       박남기   6403     월   
203  203   전탐  VI00001                        C프로그래밍       김선형   6202     화   
204  204   전탐  VI00001                        C프로그래밍       이하규   6406     월   
205  205   전탐  VI00001                        C프로그래밍       이상윤   6206     월   
255  255   전선  LD00016                       중국영화의이해       윤영도   M404     월   
260  260   전필  MA00003                       사회복지조사론       김연아   1506     화   
273  273   전필  MA00027                    사회복지현장실습 I       서진환   9999     토   
360  360   전필  PC00020             캡스톤디자인-콘텐츠제작프로젝트1    

In [70]:
room.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 51 entries, 0 to 361
Data columns (total 11 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   id          51 non-null     int64 
 1   kind        51 non-null     object
 2   code        51 non-null     object
 3   class_name  51 non-null     object
 4   prof_name   51 non-null     object
 5   room        50 non-null     object
 6   date1       50 non-null     object
 7   date2       6 non-null      object
 8   start       50 non-null     object
 9   end         50 non-null     object
 10  kwan        51 non-null     object
dtypes: int64(1), object(10)
memory usage: 4.8+ KB


In [71]:
room = room[['kwan', 'room']]
room

kwan   room
0             승연관   1505
1             월당관   3501
2            미가엘관   M301
3             월당관   3301
4            미가엘관   M406
5    성미가엘성당&피츠버그홀   9301
6            미가엘관   M407
8            새천년관   7205
9             월당관   3401
10            승연관   1503
11           미가엘관   M402
12           새천년관   7207
15            월당관   3402
18           새천년관   7204
20           새천년관   7206
21           새천년관   7202
26           새천년관   7304
27           새천년관   7305
28           새천년관   7306
29           새천년관   7307
34          정보과학관   6201
35          정보과학관   6404
36          정보과학관   6109
37          정보과학관   6405
42            승연관   1504
52           새천년관   7208
53           새천년관   7308
54           새천년관   7309
62           미가엘관   M401
64            월당관   3302
66            일만관   B105
70            월당관   3502
74          정보과학관   6110
77           미가엘관   M403
85           새천년관   7301
95           새천년관   7303
109  성미가엘성당&피츠버그홀   9101
135           나눔관  5C301
143           nan    NaN
145           승연관   1502
179          미가엘관   M205
191          새천년관   7302
197         정보과학관   6403
203         정보과학관   6202
204         정보과학관   6406
205         정보과학관   6206
255          미가엘관   M404
260           승연관   1506
273  성미가엘성당&피츠버그홀   9999
360          새천년관   7417
361         정보과학관   6402

In [80]:
room.at[273, 'room'] = np.NaN
room

kwan   room
0             승연관   1505
1             월당관   3501
2            미가엘관   M301
3             월당관   3301
4            미가엘관   M406
5    성미가엘성당&피츠버그홀   9301
6            미가엘관   M407
8            새천년관   7205
9             월당관   3401
10            승연관   1503
11           미가엘관   M402
12           새천년관   7207
15            월당관   3402
18           새천년관   7204
20           새천년관   7206
21           새천년관   7202
26           새천년관   7304
27           새천년관   7305
28           새천년관   7306
29           새천년관   7307
34          정보과학관   6201
35          정보과학관   6404
36          정보과학관   6109
37          정보과학관   6405
42            승연관   1504
52           새천년관   7208
53           새천년관   7308
54           새천년관   7309
62           미가엘관   M401
64            월당관   3302
66            일만관   B105
70            월당관   3502
74          정보과학관   6110
77           미가엘관   M403
85           새천년관   7301
95           새천년관   7303
109  성미가엘성당&피츠버그홀   9101
135           나눔관  5C301
143           nan    NaN
145           승연관   1502
179          미가엘관   M205
191          새천년관   7302
197         정보과학관   6403
203         정보과학관   6202
204         정보과학관   6406
205         정보과학관   6206
255          미가엘관   M404
260           승연관   1506
273  성미가엘성당&피츠버그홀    NaN
360          새천년관   7417
361         정보과학관   6402

In [81]:
room.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 51 entries, 0 to 361
Data columns (total 2 columns):
 #   Column  Non-Null Count  Dtype 
---  ------  --------------  ----- 
 0   kwan    51 non-null     object
 1   room    49 non-null     object
dtypes: object(2)
memory usage: 3.2+ KB


In [85]:
room = room.dropna(axis=0)
room.reset_index()
room = room[['kwan','room']]
room

[['kwan', 'room']]

In [83]:
room

kwan   room
0             승연관   1505
1             월당관   3501
2            미가엘관   M301
3             월당관   3301
4            미가엘관   M406
5    성미가엘성당&피츠버그홀   9301
6            미가엘관   M407
8            새천년관   7205
9             월당관   3401
10            승연관   1503
11           미가엘관   M402
12           새천년관   7207
15            월당관   3402
18           새천년관   7204
20           새천년관   7206
21           새천년관   7202
26           새천년관   7304
27           새천년관   7305
28           새천년관   7306
29           새천년관   7307
34          정보과학관   6201
35          정보과학관   6404
36          정보과학관   6109
37          정보과학관   6405
42            승연관   1504
52           새천년관   7208
53           새천년관   7308
54           새천년관   7309
62           미가엘관   M401
64            월당관   3302
66            일만관   B105
70            월당관   3502
74          정보과학관   6110
77           미가엘관   M403
85           새천년관   7301
95           새천년관   7303
109  성미가엘성당&피츠버그홀   9101
135           나눔관  5C301
143           nan    NaN
145           승연관   1502
179          미가엘관   M205
191          새천년관   7302
197         정보과학관   6403
203         정보과학관   6202
204         정보과학관   6406
205         정보과학관   6206
255          미가엘관   M404
260           승연관   1506
273  성미가엘성당&피츠버그홀    NaN
360          새천년관   7417
361         정보과학관   6402

In [96]:
## room class 밑에서 추가 작업 있음

room = data.drop_duplicates(['room'])
room.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 51 entries, 0 to 361
Data columns (total 11 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   id          50 non-null     float64
 1   kind        51 non-null     object 
 2   code        51 non-null     object 
 3   class_name  51 non-null     object 
 4   prof_name   51 non-null     object 
 5   room        50 non-null     object 
 6   date1       50 non-null     object 
 7   date2       6 non-null      object 
 8   start       50 non-null     object 
 9   end         50 non-null     object 
 10  kwan        51 non-null     object 
dtypes: float64(1), object(10)
memory usage: 4.8+ KB


In [97]:
room.at[273, 'kwan'] = np.NaN
room.at[273, 'room'] = np.NaN
room

id kind     code                    class_name prof_name   room date1  \
0      0.0   교필  AK00055           비아메디아채플(나와 타인 이해하기)       이남호   1505     월   
1      1.0   교필  AK00055              비아메디아채플(리더십의 영성)       김한승   3501     월   
2      2.0   교필  AK00055                비아메디아채플(서양음악사)       박태영   M301     월   
3      3.0   교필  AK00055               비아메디아채플(성공회이야기)       노철래   3301     목   
4      4.0   교필  AK00055               비아메디아채플(시각과 영성)       김돈회   M406     월   
5      5.0   교필  AK00055            비아메디아채플(열린 성공회 예배)       미지정   9301     목   
6      6.0   교필  AK00055             비아메디아채플(예술치료와 영성)       이현지   M407     월   
8      8.0   교필  AK00055         비아메디아채플(인간심리와 기독교 명상)       이종민   7205     목   
9      9.0   교필  AK00055       비아메디아채플(쟁점으로 만나는 그리스도교)       김종훈   3401     목   
10    10.0   교필  AK00055  비아메디아채플(정신분석으로 바라본 애착관계와 영성)       김미희   1503     목   
11    11.0   교필  AK00055            비아메디아채플(함께 나누는 세상)       양만호   M402     월   
12    12.0   교필  AC00003                         사회봉사I       노철래   7207   NaN   
15    15.0   교필  AC00012            대학생활세미나 I(미디어융합2반)       박남기   3402     금   
18    18.0   교필  AC00012             대학생활세미나 I(사회융합1반)       송재민   7204     금   
20    20.0   교필  AC00012             대학생활세미나 I(사회융합3반)       남일성   7206     금   
21    21.0   교필  AC00012             대학생활세미나 I(사회융합4반)       차피득   7202     금   
26    26.0   교필  AC00012             대학생활세미나 I(인문융합1반)       김혜인   7304     금   
27    27.0   교필  AC00012             대학생활세미나 I(인문융합2반)       이남주   7305     금   
28    28.0   교필  AC00012             대학생활세미나 I(인문융합3반)       이주엽   7306     금   
29    29.0   교필  AC00012             대학생활세미나 I(인문융합4반)       미지정   7307     금   
34    34.0   교필  AC00012             대학생활세미나 I(IT융합1반)       김선형   6201     금   
35    35.0   교필  AC00012             대학생활세미나 I(IT융합2반)       이하규   6404     금   
36    36.0   교필  AC00012             대학생활세미나 I(IT융합3반)       정연식   6109     금   
37    37.0   교필  AC00012             대학생활세미나 I(IT융합4반)       이상윤   6405     금   
42    42.0   교필  AC00013                           말과글       오현화   1504     수   
52    52.0   교필  AC00007                         인권과평화       이남주   7208     화   
53    53.0   교필  AC00007                         인권과평화       양경은   7308     화   
54    54.0   교필  AC00007                         인권과평화       조경희   7309     화   
62    62.0   교필  AC00015                  데이터 활용 및  분석       윤명호   M401     금   
64    64.0   교필  AC00015                  데이터 활용 및  분석       유상신   3302     금   
66    66.0   교필  AC00015                  데이터 활용 및  분석       이해신   B105     금   
70    70.0   교선  AK00025                       현대사회와법1       이기덕   3502     화   
74    74.0   교선  AE00019                      국제질서의 이해       이남주   6110     화   
77    77.0   교선  AF00056                기후위기와 지속가능한 농업       김병수   M403     목   
85    85.0   교선  AD00172         더불어숲인문학\n(삶을 위한 인문공부)       고병헌   7301     월   
95    95.0   교선  AH00020                    진로와 직업의 세계       한상진   7303     화   
109  109.0   교선  AE00099                            마음       남일성   9101     화   
135  135.0   교선  AD00164                        생활과 발레       장소정  5C301     목   
143  143.0   교선  AE00029                        사회봉사II       노철래    NaN     수   
145  145.0   평교  AD00124                       문화예술교육론       신재민   1502     금   
179  179.0   전탐  VS00012                    경제환경과 기업경영        이웅   M205     화   
191  191.0   전탐  VS00020                  전공탐색세미나(경제학)       유철규   7302     월   
197  197.0   전탐  VD00003                      디지털기초디자인       박남기   6403     월   
203  203.0   전탐  VI00001                        C프로그래밍       김선형   6202     화   
204  204.0   전탐  VI00001                        C프로그래밍       이하규   6406     월   
205  205.0   전탐  VI00001                        C프로그래밍       이상윤   6206     월   
255  255.0   전선  LD00016                       중국영화의이해       윤영도   M404     월   
260  260.0   전필  MA00003                       사회복지조사론       김연아   1506     화   
273    NaN   전필  MA00027               

In [98]:
room = room[['kwan','room']]
room = room.dropna(axis=0)
room.reset_index(inplace=True)
room = room[['kwan','room']]
room

kwan   room
0            승연관   1505
1            월당관   3501
2           미가엘관   M301
3            월당관   3301
4           미가엘관   M406
5   성미가엘성당&피츠버그홀   9301
6           미가엘관   M407
7           새천년관   7205
8            월당관   3401
9            승연관   1503
10          미가엘관   M402
11          새천년관   7207
12           월당관   3402
13          새천년관   7204
14          새천년관   7206
15          새천년관   7202
16          새천년관   7304
17          새천년관   7305
18          새천년관   7306
19          새천년관   7307
20         정보과학관   6201
21         정보과학관   6404
22         정보과학관   6109
23         정보과학관   6405
24           승연관   1504
25          새천년관   7208
26          새천년관   7308
27          새천년관   7309
28          미가엘관   M401
29           월당관   3302
30           일만관   B105
31           월당관   3502
32         정보과학관   6110
33          미가엘관   M403
34          새천년관   7301
35          새천년관   7303
36  성미가엘성당&피츠버그홀   9101
37           나눔관  5C301
38           승연관   1502
39          미가엘관   M205
40          새천년관   7302
41         정보과학관   6403
42         정보과학관   6202
43         정보과학관   6406
44         정보과학관   6206
45          미가엘관   M404
46           승연관   1506
47          새천년관   7417
48         정보과학관   6402

In [100]:
room.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 49 entries, 0 to 48
Data columns (total 2 columns):
 #   Column  Non-Null Count  Dtype 
---  ------  --------------  ----- 
 0   kwan    49 non-null     object
 1   room    49 non-null     object
dtypes: object(2)
memory usage: 912.0+ bytes


In [104]:
## floor 속성 만들기

df = room

df['floor'] = df.room.str[1:2]
df.loc[df['floor']=='C', 'floor'] = '3'
df

kwan   room floor
0            승연관   1505     5
1            월당관   3501     5
2           미가엘관   M301     3
3            월당관   3301     3
4           미가엘관   M406     4
5   성미가엘성당&피츠버그홀   9301     3
6           미가엘관   M407     4
7           새천년관   7205     2
8            월당관   3401     4
9            승연관   1503     5
10          미가엘관   M402     4
11          새천년관   7207     2
12           월당관   3402     4
13          새천년관   7204     2
14          새천년관   7206     2
15          새천년관   7202     2
16          새천년관   7304     3
17          새천년관   7305     3
18          새천년관   7306     3
19          새천년관   7307     3
20         정보과학관   6201     2
21         정보과학관   6404     4
22         정보과학관   6109     1
23         정보과학관   6405     4
24           승연관   1504     5
25          새천년관   7208     2
26          새천년관   7308     3
27          새천년관   7309     3
28          미가엘관   M401     4
29           월당관   3302     3
30           일만관   B105     1
31           월당관   3502     5
32         정보과학관   6110     1
33          미가엘관   M403     4
34          새천년관   7301     3
35          새천년관   7303     3
36  성미가엘성당&피츠버그홀   9101     1
37           나눔관  5C301     3
38           승연관   1502     5
39          미가엘관   M205     2
40          새천년관   7302     3
41         정보과학관   6403     4
42         정보과학관   6202     2
43         정보과학관   6406     4
44         정보과학관   6206     2
45          미가엘관   M404     4
46           승연관   1506     5
47          새천년관   7417     4
48         정보과학관   6402     4

In [105]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 49 entries, 0 to 48
Data columns (total 3 columns):
 #   Column  Non-Null Count  Dtype 
---  ------  --------------  ----- 
 0   kwan    49 non-null     object
 1   room    49 non-null     object
 2   floor   49 non-null     object
dtypes: object(3)
memory usage: 1.3+ KB


In [106]:
room = df
room

kwan   room floor
0            승연관   1505     5
1            월당관   3501     5
2           미가엘관   M301     3
3            월당관   3301     3
4           미가엘관   M406     4
5   성미가엘성당&피츠버그홀   9301     3
6           미가엘관   M407     4
7           새천년관   7205     2
8            월당관   3401     4
9            승연관   1503     5
10          미가엘관   M402     4
11          새천년관   7207     2
12           월당관   3402     4
13          새천년관   7204     2
14          새천년관   7206     2
15          새천년관   7202     2
16          새천년관   7304     3
17          새천년관   7305     3
18          새천년관   7306     3
19          새천년관   7307     3
20         정보과학관   6201     2
21         정보과학관   6404     4
22         정보과학관   6109     1
23         정보과학관   6405     4
24           승연관   1504     5
25          새천년관   7208     2
26          새천년관   7308     3
27          새천년관   7309     3
28          미가엘관   M401     4
29           월당관   3302     3
30           일만관   B105     1
31           월당관   3502     5
32         정보과학관   6110     1
33          미가엘관   M403     4
34          새천년관   7301     3
35          새천년관   7303     3
36  성미가엘성당&피츠버그홀   9101     1
37           나눔관  5C301     3
38           승연관   1502     5
39          미가엘관   M205     2
40          새천년관   7302     3
41         정보과학관   6403     4
42         정보과학관   6202     2
43         정보과학관   6406     4
44         정보과학관   6206     2
45          미가엘관   M404     4
46           승연관   1506     5
47          새천년관   7417     4
48         정보과학관   6402     4

In [107]:
room.to_csv('room.csv', encoding='utf-8-sig')

In [108]:
data

id kind     code           class_name prof_name  room date1 date2  \
0      0.0   교필  AK00055  비아메디아채플(나와 타인 이해하기)       이남호  1505     월   NaN   
1      1.0   교필  AK00055     비아메디아채플(리더십의 영성)       김한승  3501     월   NaN   
2      2.0   교필  AK00055       비아메디아채플(서양음악사)       박태영  M301     월   NaN   
3      3.0   교필  AK00055      비아메디아채플(성공회이야기)       노철래  3301     목   NaN   
4      4.0   교필  AK00055      비아메디아채플(시각과 영성)       김돈회  M406     월   NaN   
..     ...  ...      ...                  ...       ...   ...   ...   ...   
413  413.0   전선  XE00003             데이터기초수학2       백지현  6206     화     목   
414  414.0   전선  XE00004               고급머신러닝       홍성준  6405     목   NaN   
415  415.0   전선  XE00005        고급Python프로그래밍       홍성준  6405     화   NaN   
416  416.0   전선  XE00006             딥러닝프레임워크       이상윤  6405     월     수   
417  417.0   전선  XE00007              확률과랜덤변수       이상윤  6405     월     수   

     start    end   kwan  
0    11:00  11:50    승연관  
1    11:00  11:50    월당관  
2    11:00  11:50   미가엘관  
3    11:00  11:50    월당관  
4    11:00  11:50   미가엘관  
..     ...    ...    ...  
413  12:00  13:15  정보과학관  
414  09:00  11:50  정보과학관  
415  09:00  11:50  정보과학관  
416  13:30  14:45  정보과학관  
417  16:30  17:45  정보과학관  

[418 rows x 11 columns]

In [141]:
df = data

df = df.drop(['id'], axis=1)
df

kind     code           class_name prof_name  room date1 date2  start  \
0     교필  AK00055  비아메디아채플(나와 타인 이해하기)       이남호  1505     월   NaN  11:00   
1     교필  AK00055     비아메디아채플(리더십의 영성)       김한승  3501     월   NaN  11:00   
2     교필  AK00055       비아메디아채플(서양음악사)       박태영  M301     월   NaN  11:00   
3     교필  AK00055      비아메디아채플(성공회이야기)       노철래  3301     목   NaN  11:00   
4     교필  AK00055      비아메디아채플(시각과 영성)       김돈회  M406     월   NaN  11:00   
..   ...      ...                  ...       ...   ...   ...   ...    ...   
413   전선  XE00003             데이터기초수학2       백지현  6206     화     목  12:00   
414   전선  XE00004               고급머신러닝       홍성준  6405     목   NaN  09:00   
415   전선  XE00005        고급Python프로그래밍       홍성준  6405     화   NaN  09:00   
416   전선  XE00006             딥러닝프레임워크       이상윤  6405     월     수  13:30   
417   전선  XE00007              확률과랜덤변수       이상윤  6405     월     수  16:30   

       end   kwan  
0    11:50    승연관  
1    11:50    월당관  
2    11:50   미가엘관  
3    11:50    월당관  
4    11:50   미가엘관  
..     ...    ...  
413  13:15  정보과학관  
414  11:50  정보과학관  
415  11:50  정보과학관  
416  14:45  정보과학관  
417  17:45  정보과학관  

[418 rows x 10 columns]

In [142]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 418 entries, 0 to 417
Data columns (total 10 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   kind        418 non-null    object
 1   code        418 non-null    object
 2   class_name  418 non-null    object
 3   prof_name   418 non-null    object
 4   room        416 non-null    object
 5   date1       416 non-null    object
 6   date2       105 non-null    object
 7   start       416 non-null    object
 8   end         416 non-null    object
 9   kwan        418 non-null    object
dtypes: object(10)
memory usage: 32.8+ KB


In [143]:
df['room'].value_counts()

7208     18
M402     18
6201     17
7304     15
6202     14
7309     14
7204     14
7206     14
7205     13
7207     13
6406     12
6403     12
7308     12
3402     12
7202     11
1505     11
6206     10
6405     10
M406     10
M407     10
6110     10
M301      9
3301      9
7302      9
M401      9
3502      8
7306      8
6404      8
1504      8
7305      8
1503      7
M403      7
3401      6
1506      5
1502      5
3302      5
6109      5
7301      4
M404      4
B105      3
5C301     3
M205      3
9301      3
7307      2
3501      2
7303      2
9101      1
9999      1
7417      1
6402      1
Name: room, dtype: int64

In [144]:
df[df['room'] == '9999']

kind     code  class_name prof_name  room date1 date2  start    end  \
273   전필  MA00027  사회복지현장실습 I       서진환  9999     토   NaN  09:00  11:50   

             kwan  
273  성미가엘성당&피츠버그홀

In [145]:
df.loc[273, 'room'] = np.NaN
df.loc[273, 'kwan'] = np.NaN
df[273:274]

kind     code  class_name prof_name room date1 date2  start    end kwan
273   전필  MA00027  사회복지현장실습 I       서진환  NaN     토   NaN  09:00  11:50  NaN

In [146]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 418 entries, 0 to 417
Data columns (total 10 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   kind        418 non-null    object
 1   code        418 non-null    object
 2   class_name  418 non-null    object
 3   prof_name   418 non-null    object
 4   room        415 non-null    object
 5   date1       416 non-null    object
 6   date2       105 non-null    object
 7   start       416 non-null    object
 8   end         416 non-null    object
 9   kwan        417 non-null    object
dtypes: object(10)
memory usage: 32.8+ KB


In [147]:
df['kwan'].value_counts()

새천년관            158
정보과학관            99
미가엘관             70
월당관              42
승연관              36
성미가엘성당&피츠버그홀      4
일만관               3
나눔관               3
nan               2
Name: kwan, dtype: int64

In [148]:
df[df['kwan'] == 'nan']

kind     code class_name prof_name room date1 date2  start    end kwan
143   교선  AE00029     사회봉사II       노철래  NaN     수   NaN  10:30  11:45  nan
144   교선  AE00030    사회봉사III       노철래  NaN     목   NaN  10:30  11:45  nan

In [149]:
df.loc[143, 'kwan'] = np.NaN
df.loc[144, 'kwan'] = np.NaN
df[143:145]

kind     code class_name prof_name room date1 date2  start    end kwan
143   교선  AE00029     사회봉사II       노철래  NaN     수   NaN  10:30  11:45  NaN
144   교선  AE00030    사회봉사III       노철래  NaN     목   NaN  10:30  11:45  NaN

In [150]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 418 entries, 0 to 417
Data columns (total 10 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   kind        418 non-null    object
 1   code        418 non-null    object
 2   class_name  418 non-null    object
 3   prof_name   418 non-null    object
 4   room        415 non-null    object
 5   date1       416 non-null    object
 6   date2       105 non-null    object
 7   start       416 non-null    object
 8   end         416 non-null    object
 9   kwan        415 non-null    object
dtypes: object(10)
memory usage: 32.8+ KB


In [151]:
df[df['room'].isnull()]

kind     code  class_name prof_name room date1 date2  start    end kwan
143   교선  AE00029      사회봉사II       노철래  NaN     수   NaN  10:30  11:45  NaN
144   교선  AE00030     사회봉사III       노철래  NaN     목   NaN  10:30  11:45  NaN
273   전필  MA00027  사회복지현장실습 I       서진환  NaN     토   NaN  09:00  11:50  NaN

In [152]:
df

kind     code           class_name prof_name  room date1 date2  start  \
0     교필  AK00055  비아메디아채플(나와 타인 이해하기)       이남호  1505     월   NaN  11:00   
1     교필  AK00055     비아메디아채플(리더십의 영성)       김한승  3501     월   NaN  11:00   
2     교필  AK00055       비아메디아채플(서양음악사)       박태영  M301     월   NaN  11:00   
3     교필  AK00055      비아메디아채플(성공회이야기)       노철래  3301     목   NaN  11:00   
4     교필  AK00055      비아메디아채플(시각과 영성)       김돈회  M406     월   NaN  11:00   
..   ...      ...                  ...       ...   ...   ...   ...    ...   
413   전선  XE00003             데이터기초수학2       백지현  6206     화     목  12:00   
414   전선  XE00004               고급머신러닝       홍성준  6405     목   NaN  09:00   
415   전선  XE00005        고급Python프로그래밍       홍성준  6405     화   NaN  09:00   
416   전선  XE00006             딥러닝프레임워크       이상윤  6405     월     수  13:30   
417   전선  XE00007              확률과랜덤변수       이상윤  6405     월     수  16:30   

       end   kwan  
0    11:50    승연관  
1    11:50    월당관  
2    11:50   미가엘관  
3    11:50    월당관  
4    11:50   미가엘관  
..     ...    ...  
413  13:15  정보과학관  
414  11:50  정보과학관  
415  11:50  정보과학관  
416  14:45  정보과학관  
417  17:45  정보과학관  

[418 rows x 10 columns]